packages/core/admin/ee/server/migrations/review-workflows-stage-attribute.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const semver = require('semver');
+const { getOr } = require('lodash/fp');
+const { mapAsync } = require('@strapi/utils');
+const { STAGE_MODEL_UID } = require('../constants/workflows');
+const { findTables } = require('../utils/persisted-tables');
+
+function checkVersionThreshold(startVersion, currentVersion, thresholdVersion) {
+  return semver.gte(currentVersion, thresholdVersion) && semver.lt(startVersion, thresholdVersion);
+}
+
+async function migrateStageAttribute({ oldContentTypes, contentTypes }) {
+  const getRWVersion = getOr('0.0.0', `${STAGE_MODEL_UID}.options.version`);
+  const oldRWVersion = getRWVersion(oldContentTypes);
+  const currentRWVersion = getRWVersion(contentTypes);
+
+  const migrationNeeded = checkVersionThreshold(oldRWVersion, currentRWVersion, '1.1.0');
+
+  if (migrationNeeded) {
+    const oldAttributeTableName = 'strapi_review_workflows_stage';
+    const newAttributeTableName = 'strapi_stage';
+    const tables = await findTables({ strapi }, new RegExp(oldAttributeTableName));
+
+    await mapAsync(tables, async (tableName) => {
+      const newTableName = tableName.replace(oldAttributeTableName, newAttributeTableName);
```

Comment:
Using raw table names can be a little bit brittle, specially knowing we will change the way table names are generated in the future (for long table names).

I don't know if we already have a method to get table names from a content type attribute, but it might help prevent issues in the future here.

Reply:
I think you are right here.
We might need to centralize everything DB related inside our database package.
We already have some "low level" table name retrieval functions.
packages/core/database/lib/metadata/relations.js:19

But it also needs the table name, which is resolved in database functions inside the metadata of the content-types..
So we do not have "easy" way to manage this.

Maybe we can think of adding new functions to the database package. Also we might want to automatically manage this, and for example, store patch of changes on content-type to be able to migrate attribute that changes their name 🤔 (because if we change the name of an attr for now, we just lost the data 😅 )

Reply:
ah you are right 🤔 this migration files are executed from the strapi package and not the database. It's tricky because migrations should be on the database layer.

Indeed there doesnt seem to be an easy way to transform this to table names.

I see the table name is just the collectionName https://github.com/strapi/strapi/blob/main/packages/core/database/lib/utils/content-types.js#L20
and maybe we could also use the retrieval functions you mention + this one. Nevertheless, It still looks brittle and we might want to think this on a broader scale.

The renaming attributes might be a topic indeed, and for sure we also will be discussing it during Content History.